DataFormats/HcalDigi/interface/HODataFrame.h

```diff
@@ -17,40 +16,77 @@ class HODataFrame {
 public:
   typedef HcalDetId key_type; ///< For the sorted collection
 
-  HODataFrame(); // for persistence
-  explicit HODataFrame(const HcalDetId& id);
+  HODataFrame() 
+    : id_(0), size_(0), hcalPresamples_(0)
+  {}
+  explicit constexpr HODataFrame(const HcalDetId& id) 
+    : id_(id), size_(0), hcalPresamples_(0)
+  {// TODO : test id for HcalOuter
+  }
   
-  const HcalDetId& id() const { return id_; }
-  const HcalElectronicsId& elecId() const { return electronicsId_; }
+  constexpr HcalDetId const& id() const { return id_; }
+  constexpr HcalElectronicsId const& elecId() const { return electronicsId_; }
   
   /// total number of samples in the digi     
-  int size() const { return size_&0xF; }
+  constexpr int size() const { return size_&0xF; }
   /// number of samples before the sample from the triggered beam crossing (according to the hardware)
-  int presamples() const { return hcalPresamples_&0xF; }
+  constexpr int presamples() const { return hcalPresamples_&0xF; }
    /// was ZS MarkAndPass?
-  bool zsMarkAndPass() const { return (hcalPresamples_&0x10); }
+  constexpr bool zsMarkAndPass() const { return (hcalPresamples_&0x10); }
   /// was ZS unsuppressed?
-  bool zsUnsuppressed() const { return (hcalPresamples_&0x20); }
+  constexpr bool zsUnsuppressed() const { return (hcalPresamples_&0x20); }
   /// zs crossing mask (which sums considered)
-  uint32_t zsCrossingMask() const { return (hcalPresamples_&0x3FF000)>>12; }
+  constexpr uint32_t zsCrossingMask() const { return (hcalPresamples_&0x3FF000)>>12; }
  
   /// access a sample
-  const HcalQIESample& operator[](int i) const { return data_[i]; }
+  constexpr const HcalQIESample& operator[](int i) const { return data_[i]; }
   /// access a sample
-  const HcalQIESample& sample(int i) const { return data_[i]; }
+  constexpr const HcalQIESample& sample(int i) const { return data_[i]; }
  
   /// offset of bunch number for this channel relative to nominal set in the unpacker (range is +7->-7.  -1000 indicates the data is invalid/unavailable)
-  int fiberIdleOffset() const;
+  constexpr int fiberIdleOffset() const {
+    int val=(hcalPresamples_&0xF00)>>8;
+    return (val==0)?(-1000):(((val&0x8)==0)?(-(val&0x7)):(val&0x7));
```

Comment:
Here and in some other methods, could you use some symbolic names for the constants `0xF00`, `0x7`, `0x8` etc ?

I think they were already like this in the original code,  but it could be an occasion to improve upon it...